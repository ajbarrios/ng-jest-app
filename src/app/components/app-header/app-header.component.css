.header-wrapper {
  display: flex;
  justify-content: space-between;
  padding: 0.5rem;
  background-color: #282a35;
  color: #fff;
}

.title {
  margin: 0.5rem;
}

.start {
  display: flex;
  align-items: center;
  gap: 1rem;
}

.end {
  display: flex;
  align-items: center;
  gap: 1rem;
}
