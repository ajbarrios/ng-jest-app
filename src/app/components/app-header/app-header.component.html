<header class="header-wrapper">
  <div class="start">
    <h2 class="title">{{ title }}</h2>
    <ng-content select="[start]"></ng-content>
  </div>
  <div class="end">
    <ng-content select="[end]"></ng-content>
  </div>
</header>
