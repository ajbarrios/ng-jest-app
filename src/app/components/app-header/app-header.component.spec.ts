import { Component } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';

import { AppHeaderComponent } from './app-header.component';

describe('AppHeaderComponent', () => {
  let component: AppHeaderComponent;
  let fixture: ComponentFixture<AppHeaderComponent>;

  @Component({
    selector: 'app-header-test',
    template: ` <app-header [title]="title">
      <nav start class="navigation">
        <a class="item" href="https://www.w3schools.com/html/" target="_blank"
          >HTML</a
        >
        <a class="item" href="https://www.w3schools.com/css/" target="_blank"
          >CSS</a
        >
        <a class="item" href="https://www.w3schools.com/js/" target="_blank"
          >JavaScript</a
        >
        <a class="item" href="https://www.w3schools.com/python/" target="_blank"
          >Python</a
        >
      </nav>
      <button end>Login</button>
    </app-header>`,
  })
  class AppHeaderTestComponent extends AppHeaderComponent {}

  @Component({
    selector: 'app-header-test-2',
    template: `<app-header title="just a title">
    </app-header>`,
  })
  class AppHeaderTestComponent2 extends AppHeaderComponent {}

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [
        AppHeaderComponent,
        AppHeaderTestComponent,
        AppHeaderTestComponent2,
      ],
    }).compileComponents();

    fixture = TestBed.createComponent(AppHeaderTestComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  afterEach(() => {
    fixture.destroy();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render default title if not provided', () => {
    fixture.detectChanges();
    const title = fixture.nativeElement.querySelector('.title') as HTMLElement;
    expect(title.textContent).toMatchSnapshot();
  });

  it('should render a provided title', () => {
    const expectedTitle = 'just a title';
    component.title = expectedTitle;
    fixture.detectChanges();
    const title = fixture.nativeElement.querySelector('.title') as HTMLElement;
    expect(title).toMatchSnapshot();
  });

  it('should render slotted content in the start slot', () => {
    fixture.detectChanges();
    const startContent = fixture.nativeElement.querySelector(
      '[start]'
    ) as HTMLElement;
    expect(startContent).toMatchSnapshot();
  });

  it('should render slotted content in the end slot', () => {
    fixture.detectChanges();
    const endContent = fixture.nativeElement.querySelector(
      '[end]'
    ) as HTMLElement;
    expect(endContent).toMatchSnapshot();
  });

  it('should not render slotted content in the start slot', () => {
    fixture = TestBed.createComponent(AppHeaderTestComponent2);
    component = fixture.componentInstance;
    fixture.detectChanges();
    const startContent = fixture.nativeElement.querySelector(
      '[start]'
    ) as HTMLElement;
    expect(startContent).toMatchSnapshot();
  });

  it('should not render slotted content in the end slot', () => {
    fixture = TestBed.createComponent(AppHeaderTestComponent2);
    component = fixture.componentInstance;
    fixture.detectChanges();
    const startContent = fixture.nativeElement.querySelector(
      '[end]'
    ) as HTMLElement;
    expect(startContent).toMatchSnapshot();
  });
});
