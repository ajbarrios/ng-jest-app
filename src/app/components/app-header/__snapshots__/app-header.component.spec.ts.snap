// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeaderComponent should not render slotted content in the end slot 1`] = `null`;

exports[`AppHeaderComponent should not render slotted content in the start slot 1`] = `null`;

exports[`AppHeaderComponent should render a provided title 1`] = `
<h2
  class="title"
>
  just a title
</h2>
`;

exports[`AppHeaderComponent should render default title if not provided 1`] = `""`;

exports[`AppHeaderComponent should render slotted content in the end slot 1`] = `
<button
  end=""
>
  Login
</button>
`;

exports[`AppHeaderComponent should render slotted content in the start slot 1`] = `
<nav
  class="navigation"
  start=""
>
  <a
    class="item"
    href="https://www.w3schools.com/html/"
    target="_blank"
  >
    HTML
  </a>
  <a
    class="item"
    href="https://www.w3schools.com/css/"
    target="_blank"
  >
    CSS
  </a>
  <a
    class="item"
    href="https://www.w3schools.com/js/"
    target="_blank"
  >
    JavaScript
  </a>
  <a
    class="item"
    href="https://www.w3schools.com/python/"
    target="_blank"
  >
    Python
  </a>
</nav>
`;
